### ISSUE TITLE
error when including more than one listener in the same screen 

### ISSUE THREAD

**Issue by davidfabbretti** [orchidsoftware/platform#2489]
**Describe the bug**

when more than one listener is included in the same screen when trying to access $this->query from the second listener, this variable returns null.

**Screenshots**
![WhatsApp Image 2023-01-02 at 12 44 30](https://user-images.githubusercontent.com/17312127/210255153-5147ba35-19f7-4005-862f-e2fec5c82619.jpeg)

![WhatsApp Image 2023-01-02 at 12 44 31](https://user-images.githubusercontent.com/17312127/210255194-d5e6243f-b6a1-46e9-8c42-16889a685e7c.jpeg)

![WhatsApp Image 2023-01-02 at 12 44 31-2](https://user-images.githubusercontent.com/17312127/210255229-86668d8c-b0f3-4c0c-bb76-fb7df9a1524d.jpeg)

![WhatsApp Image 2023-01-02 at 12 36 27](https://user-images.githubusercontent.com/17312127/210255249-2e4abbe7-e276-4700-94db-5ce3a8f9d720.jpeg)

**Comment by tabuna** [comment]
I can confirm this problem. This is because we can have different levels of nesting. For example:

```php
public function layout(): iterable
{
    return [
        Layout::columns([
            AmountListener::class,
            //...
        ]),

        Layout::columns([
            AmountListener2::class,
            //...
        ]),
    ];
}
```

To understand which layer to return, we go through all of them superficially.

**Comment by tabuna** [comment]
This problem should be solved with the [release of 13.9.3](https://github.com/orchidsoftware/platform/releases/tag/13.9.3).
Feel free to reopen this if there are problems.
